### DISCUSSION TITLE
OpenSSL 3.0 LTS support

### DISCUSSION THREAD

**Discussion by GrahamCampbell** [cachewerk/relay#71] [Support]
The extension links against OpenSSL 1.1.1. OpenSSL will be EOL in a few months. Is OpenSSL 3.0.8 compatibility on the roadmap?

**Comment by michael-grunder** [comment]
Relay is already compatible with OpenSSL3.  You can find some examples (select Debian/Ubuntu) [here](https://relay.so/builds).

One thing to consider, however, is that when running Relay that links against OpenSSL3 it's essential that PHP itself was also built against OpenSSL3.

**Comment by tillkruss** [comment]
@GrahamCampbell: For which OS?

**Reply by GrahamCampbell** [reply]
AL2 with openssl 3.0.8 built by hand (https://github.com/brefphp/aws-lambda-layers/blob/baad0d8f35d41b62d6d7d76d90199e1b84649ed0/php-82/Dockerfile#L72C1-L107). Related: https://github.com/brefphp/extra-php-extensions/pull/448.

**Reply by GrahamCampbell** [reply]
The same issue will also affect Laravel Vapor users very shortly, when the same change gets rolled out to the Vapor Runtimes.

**Reply by michael-grunder** [reply]
It's just a matter of putting together the builds.  We should probably do it anyway given the EOL issue.  From the OpenSSL docs they seem to be using _mostly_ semantic versioning so building against 3.x.x should work for a long time.
